sentry-android-core/src/main/java/io/sentry/android/core/internal/util/AndroidConnectionStatusProvider.java

```diff
@@ -41,7 +44,7 @@ public AndroidConnectionStatusProvider(
     this.context = ContextUtils.getApplicationContext(context);
     this.logger = logger;
     this.buildInfoProvider = buildInfoProvider;
-    this.registeredCallbacks = new HashMap<>();
+    this.connectionStatusObservers = new ArrayList<>();
```

Comment:
just thinking if it's better to use CopyOnWrite here to avoid potential lock contention?

Reply:
I think it's fine, I'm using a lock mechanism for safe access and there aren't many writes to this list: In my testing there were 3 listeners attached.